Utilities/Thread.cpp

```diff
@@ -1409,7 +1409,7 @@ bool handle_access_violation(u32 addr, bool is_writing, x64_context* context) no
 			return false;
 		}
 
-		if (area->flags & 0x100 || (is_writing && vm::check_addr(addr)))
```

Comment:
What does it have to do with the debugger?

Reply:
It's not but it's an important bugfix when recovering from access violations on vm::main, vm::stack, vm::spu etc.

Reply:
Why checking flags in the first place? Why checking is_writing as well? Since you switch to checking only "allocated" flag, the presence of other checks on this line needs good explanation. From what I see these checks could be removed.

Reply:
Added comment and removed is_writing check,